From ea4fd0ef1b8b2da810c5012c8a2efbc6573e5dfb Mon Sep 17 00:00:00 2001
From: Ahmad Samir <a.samirh78@gmail.com>
Date: Sun, 24 Aug 2025 15:51:45 +0300
Subject: [PATCH 100/553] tst_qmetamethod: don't use indexOfSlot() with
 non-slots

At the moment, QMetaOjbect::indexOfSlot() works for both Q_SLOTS: and
invokables, but technically for a Q_INVOKABLE method that isn't under
Q_SLOTS:, we should use indexOfMethod().

Pick-to: 6.9 6.8 6.5
Change-Id: Iabf14f234a119b0f06ef078ed080e717670c6a47
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
(cherry picked from commit 4fbd808b4ce38e96bf83812e9483a0d5287f17e3)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 tests/auto/corelib/kernel/qmetamethod/tst_qmetamethod.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/auto/corelib/kernel/qmetamethod/tst_qmetamethod.cpp b/tests/auto/corelib/kernel/qmetamethod/tst_qmetamethod.cpp
index 190b8fabdaf..1674746af06 100644
--- a/tests/auto/corelib/kernel/qmetamethod/tst_qmetamethod.cpp
+++ b/tests/auto/corelib/kernel/qmetamethod/tst_qmetamethod.cpp
@@ -867,14 +867,14 @@ void tst_QMetaMethod::isConst()
     auto mo = MethodTestObject::staticMetaObject;
     {
         const auto normalized = QMetaObject::normalizedSignature("qrealInvokable()");
-        const int idx = mo.indexOfSlot(normalized);
+        const int idx = mo.indexOfMethod(normalized);
         QMetaMethod mm = mo.method(idx);
         QVERIFY(mm.isValid());
         QCOMPARE(mm.isConst(), false);
     }
     {
         const auto normalized = QMetaObject::normalizedSignature("voidInvokable()");
-        const int idx = mo.indexOfSlot(normalized);
+        const int idx = mo.indexOfMethod(normalized);
         QMetaMethod mm = mo.method(idx);
         QVERIFY(mm.isValid());
         QCOMPARE(mm.isConst(), true);
-- 
2.51.2

